test/int/framework/framework.react.test.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/*
+ * Integration tests for the React framework
+ */
+
+import * as path from 'path';
+import * as testSetup from '../testSetup';
+import { setBreakpoint, setConditionalBreakpoint } from '../intTestSupport';
+import { puppeteerSuite, puppeteerTest } from '../puppeteer/puppeteerSuite';
+import { FrameworkTestSuite } from './frameworkCommonTests';
+import { TestProjectSpec } from './frameworkTestSupport';
+
+const DATA_ROOT = testSetup.DATA_ROOT;
+const REACT_PROJECT_ROOT = path.join(DATA_ROOT, 'react');
+const TEST_SPEC = new TestProjectSpec(REACT_PROJECT_ROOT);
+
+puppeteerSuite('React Framework Tests', TEST_SPEC, (suiteContext) => {
```

Comment:
What are the advantages of using puppeteer vs what we already do?

Reply:
Basically just making it easier to write tests that interact with the page (e.g. click a button and wait for something to happen).

Reply:
The goal is just to get some code to run right? For other tests I've just used setInterval or global code. You could even ask the debug adapter to evaluate something which would trigger the code that you want to run.

Also if we use Puppeteer then the debug adapter is no longer launching the process right? That should be tested too.

Reply:
> The goal is just to get some code to run right? For other tests I've just used setInterval or global code. You could even ask the debug adapter to evaluate something which would trigger the code that you want to run.

Wouldn't doing something like eval have a slightly different impact on things like sourcemaps? Interacting with the DOM closer simulates a real use case. I can check with the team to see if there was a specific scenario they had in mind with puppeteer.

> Also if we use Puppeteer then the debug adapter is no longer launching/managing the chrome process right? That should be tested too.

Nope! The way I've got it set up right now is that we use the debug adapter to launch, and just attach puppeteer to the running instance.

Reply:
Oh, I see. It still seems like it could be overkill just to trigger some code? But if you think it will make things easier I think that's ok.